docs/concepts/cluster-administration/cloud-providers.md

```diff
@@ -58,50 +58,108 @@ The information for the annotations for AWS is taken from the comments on [aws.g
-This is an example of a minimal configuration that touches the values that most often need to be set:
+### Typical configuration
+This is an example of a typical configuration that touches the values that most
+often need to be set. It points the provider at the OpenStack cloud's Keystone
+endpoint, provides details for how to authenticate with it, and configures the
+load balancer:
 
 ```yaml
 [Global]
-username=user  
-password=pass  
-auth-url=https://<keystone_ip>/identity/v3  
-tenant-id=c869168a828847f39f7f06edd7305637  
+username=user
+password=pass
+auth-url=https://<keystone_ip>/identity/v3
+tenant-id=c869168a828847f39f7f06edd7305637
 domain-id=2a73b8f597c04551a0fdc8e95544be8a
 
 [LoadBalancer]
-subnet-id=6937f8fa-858d-4bc9-a3a5-18d2c957166a  
+subnet-id=6937f8fa-858d-4bc9-a3a5-18d2c957166a
 ```
 
 #### Global
-* `username`: Refers to the username of a valid user set in keystone.
-* `password`:Refers to the password of a valid user set in keystone.
-* `auth-url`: The URL of the keystone API used to authenticate. On OpenStack control panels, this can be found at Access and Security > API Access > Credentials.
-* `tenant-id`: Used to specify the id of the project where you want to create your resources.
-* `domain-id`: Used to specify the id of the domain your user belongs to.
+These configuration options for the OpenStack provider pertain to its global
+configuration and should appear in the `[Global]` section of the `cloud.conf`
+file:
 
-####  Load Balancer
-* `subnet-id`: Used to specify the id of the subnet you want to create your loadbalancer on. Can be found at Network > Networks. Click on the respective network to get its subnets.
+* `username` (Required): Refers to the username of a valid user set in keystone.
+* `password` (Required): Refers to the password of a valid user set in keystone.
+* `auth-url` (Required): The URL of the keystone API used to authenticate. On OpenStack control panels, this can be found at Access and Security > API Access > Credentials.
+* `tenant-id` (Required): Used to specify the id of the project where you want to create your resources.
+* `domain-id` (Optional): Used to specify the id of the domain your user belongs to.
 
-### Optional configuration
+####  Load Balancer
+These configuration options for the OpenStack provider pertain to the load
+balancer and should appear in the `[LoadBalancer]` section of the `cloud.conf`
+file:
+
+* `lb-version` (Optional): Used to override automatic version detection. Valid
+  values are `v1` or `v2`. Where no value is provided automatic detection will
+  select the highest supported version exposed by the underlying OpenStack
+  cloud.
+* `subnet-id` (Optional): Used to specify the id of the subnet you want to
+  create your loadbalancer on. Can be found at Network > Networks. Click on the
+  respective network to get its subnets.
+* `floating-network-id` (Optional): If specified, will create a floating IP for
+  the load balancer.
+* `lb-method`:
```

Comment:
lb-method: The algorithm used to distribute load between the members of the pool. The value can be: ROUND_ROBIN (default), LEAST_CONNECTIONS or SOURCE_IP